### PR TITLE
feat: optional replication

### PR DESCRIPTION
**Pull request by hamid-moghadam**
Make "nominatim replication" optional in order to use it in offline servers.
Related to #257

### CODE REVIEW

File: 3.7/config.sh [hunk 1]
```diff
@@ -8,21 +8,17 @@ if [[ "$PBF_URL" = "" && "$PBF_PATH" = "" ]]  ||  [[ "$PBF_URL" != "" && "$PBF_P
...
-    sed -i "s|__REPLICATION_URL__|$REPLICATION_URL|g" ${CONFIG_FILE}
+if [ "$REPLICATION_URL" != "" ]; then
+    sed -i "s|__REPLICATION_URL__|$REPLICATION_URL|g" ${CONFIG_FILE}    
 fi
 
 # Use the specified replication update and recheck interval values if either or both are numbers, or use the default values
 
 reg_num='^[0-9]+$'
-if [[ $REPLICATION_UPDATE_INTERVAL =~ $reg_num ]]; then
+if [[ $REPLICATION_UPDATE_INTERVAL =~ $reg_num && "$REPLICATION_URL" != "" ]]; then
```
Comment (leonardehrenfried) [inline 3.7/config.sh:18]: Could you add this dependency to the documentation or even better throw an error when an invalid combination was used?

File: 3.7/config.sh [hunk 2]
```diff
@@ -8,21 +8,17 @@ if [[ "$PBF_URL" = "" && "$PBF_PATH" = "" ]]  ||  [[ "$PBF_URL" != "" && "$PBF_P
...
     sed -i "s/NOMINATIM_REPLICATION_UPDATE_INTERVAL=86400/NOMINATIM_REPLICATION_UPDATE_INTERVAL=$REPLICATION_UPDATE_INTERVAL/g" ${CONFIG_FILE}
 fi
-if [[ $REPLICATION_RECHECK_INTERVAL =~ $reg_num ]]; then
+if [[ $REPLICATION_RECHECK_INTERVAL =~ $reg_num && "$REPLICATION_URL" != "" ]]; then
```
Comment (leonardehrenfried) [inline 3.7/config.sh:21]: Same here.

**Review by leonardehrenfried — CHANGES_REQUESTED** [review]

**Comment by leonardehrenfried** [comment]
This looks good - thanks!

@philipkozeny Do you have an opinion on this?

**Comment by philipkozeny** [comment]
An additional test case without replication would be nice. Can you add that to ```.github/workflows/ci.yml``` please?

**Comment by leonardehrenfried** [comment]
Superseded by #270

**Pull request closed without merging**